index.bs

```diff
@@ -455,7 +455,7 @@ To <dfn>sendDatagrams</dfn>, given a {{WebTransport}} object |transport| and a
 1. Let |maxSize| be |datagrams|.{{[[OutgoingMaxDatagramSize]]}}.
 1. While |queue| is not empty:
   1. Let |bytes|, |timestamp| and |promise| be |queue|'s first element.
-  1. If |bytes|'s length ≤ |maxSize|:
+  1. If |bytes|'s length > 0 and ≤ |maxSize|:
```

Comment:
This appears to just skip datagrams if they are the wrong size.  Why is the promise not rejected?

Reply:
Are you referring to ≥ maxSize, zero or both?

Re ≥ maxSize that's how it is currently specified [and](https://searchfox.org/mozilla-central/rev/4c065f1df299065c305fb48b36cdae571a43d97c/testing/web-platform/tests/webtransport/datagrams.https.any.js#204-205,208) [implemented](https://wpt.fyi/results/webtransport/datagrams.https.any.html).

Reply:
Both, in this case.  I'm not seeing that in this part of the specification.  (Good that it is implemented that way.)  Is that specified somewhere else?

This loop seems to iterate over the queue, sending any packet that is an OK size.  Once it can't send, it breaks out of the loop and removes the first item from the queue (not those that it sent, just the first one; it also doesn't  check that the first packet was sent or not).  I don't understand how that is even approximately correct.

Reply:
> Once it can't send, it breaks out of the loop and removes the first item from the queue

It breaks out of the loop, not the if-statement.

<img width="694" alt="image" src="https://github.com/user-attachments/assets/d5f206be-7c9c-4924-8f48-c6546def7928" />

Reply:
Eww.  Still, I guess that's OK.  We just silently drop things that are the wrong size.  Though I suspect that maybe this is wrong, now that we're talking about empty datagrams being valid.